src/Client.ts

```diff
@@ -5,22 +5,46 @@ import {
 } from '@kontent-ai/delivery-sdk';
 import packageInfo from '../package.json';
 import { selectedProjectCookieName } from './const';
-import { defaultProjectId } from './Utilities/SelectedProject';
+import validator from 'validator';
 
 const sourceTrackingHeaderName = 'X-KC-SOURCE';
 
-// environment variables
-const projectId = process.env.REACT_APP_PROJECT_ID || '';
 const previewApiKey = process.env.REACT_APP_PREVIEW_API_KEY || '';
 
 const cookies = new Cookies(document.cookie);
 
-let currentProjectId = projectId || cookies.get(selectedProjectCookieName);
-if (currentProjectId) {
-  cookies.set(selectedProjectCookieName, currentProjectId, { path: '/' });
-} else {
-  currentProjectId = defaultProjectId;
-}
+const getProjectIdFromEnvironment = (): string | null | undefined => {
+  const projectIdFromEnv = process.env.REACT_APP_PROJECT_ID;
+
+  if (projectIdFromEnv === undefined) {

```

Comment:
This `if` is unnecessary, because in the error-checking `if` below you check that the id is defined and in the end, you return it the same way you return it (`undefined`) here.